From f634481e940421020e52f511c1fb34aac1db4b2f Mon Sep 17 00:00:00 2001
From: Harish Mahendrakar <harish.mahendrakar@ittiam.com>
Date: Thu, 15 Dec 2016 18:04:53 +0530
Subject: Decoder: Initialize default reference buffers for all pictures

Reference buffer is now initialized to default value for
each pic before decoding the first slice in the pic

Bug: 34097866
Change-Id: Id64b123af2188217ce833f11db0e6c0681d41dfd
---
 decoder/ih264d_parse_slice.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/decoder/ih264d_parse_slice.c b/decoder/ih264d_parse_slice.c
index 880f3a8..11e2c7f 100644
--- a/decoder/ih264d_parse_slice.c
+++ b/decoder/ih264d_parse_slice.c
@@ -439,7 +439,7 @@ WORD32 ih264d_start_of_pic(dec_struct_t *ps_dec,
         ps_cur_pic->pu1_col_zero_flag = (UWORD8 *)ps_col_mv->pv_col_zero_flag;
         ps_cur_pic->ps_mv = (mv_pred_t *)ps_col_mv->pv_mv;
         ps_dec->au1_pic_buf_ref_flag[cur_pic_buf_id] = 0;
-        if(ps_dec->u1_first_slice_in_stream)
+
         {
             /*make first entry of list0 point to cur pic,so that if first Islice is in error, ref pic struct will have valid entries*/
             ps_dec->ps_ref_pic_buf_lx[0] = ps_dec->ps_dpb_mgr->ps_init_dpb[0];
-- 
cgit v1.1

